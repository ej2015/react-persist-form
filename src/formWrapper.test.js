import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import formWrapperCreater from './formWrapper'

Enzyme.configure({ adapter: new Adapter() })

const Form = ({ error, data, handleChange, handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <input
      label='Email'
      name='email'
      onChange={handleChange}
    />
    {error && <div> error </div>}
    <button type='submit'>Submit</button>
  </form>
)

const FormWrapper = formWrapperCreater({ Form })

const ConnectedForm = ({ error, handleSubmit }) => (
  <FormWrapper
    data = {{ email: '', error }}
    onSubmit={handleSubmit}
  />
)

let handleSubmit = jest.fn()

describe('formWrapperCreater', () => {
  it('renders', () => {
    const component = renderer.create(
      ConnectedForm({ handleSubmit: handleSubmit })
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should handle submit', () => {
    let event = { target: { value: 'a' } }
    let wrapper = shallow(<ConnectedForm handleSubmit={handleSubmit} />)
    wrapper.simulate('submit', event)
    expect(handleSubmit.mock.calls.length).toBe(1)
  })
})
