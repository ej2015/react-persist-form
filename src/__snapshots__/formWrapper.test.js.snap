// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formWrapperCreater renders 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    label="Email"
    name="email"
    onChange={[Function]}
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
